skimage/io/sift.py

```diff
@@ -62,9 +60,10 @@ def _sift_read(f, mode='SIFT'):
 
     data = np.fromfile(f, sep=' ')
     if data.size != nr_features * datatype.itemsize / np.dtype(float).itemsize:
-        raise IOError("Invalid %s feature file." % mode)
+        raise IOError("Invalid {} feature file.".format(mode))
 
-    f.close()
+    if f_is_str:
+        f.close()
```

Comment:
I don't really know anything about the topic of this PR, but shouldn't this be `if not f_is_str: f.close()`? If that's not the case then perhaps there should be some refactoring for clarity.